src/hacbs/components/Commits/CommitsListView.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import { useNavigate } from 'react-router-dom';
+import {
+  Button,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateSecondaryActions,
+  EmptyStateVariant,
+  PageSection,
+  PageSectionVariants,
+  SearchInput,
+  Title,
+  Text,
+  Toolbar,
+  ToolbarContent,
+  ToolbarGroup,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons/dist/js/icons';
+import { Table } from '../../../shared';
+import { Commit } from '../../types';
+import CommitsListHeader from './CommitsListHeader';
+import CommitsListRow from './CommitsListRow';
+
+interface CommitsListViewProps {
+  commits: Commit[];
+  recentOnly?: boolean;
+  applicationName?: string;
+}
+
+const CommitsListView: React.FC<CommitsListViewProps> = ({
+  commits,
+  recentOnly = false,
+  applicationName,
+}) => {
+  const navigate = useNavigate();
+  const [nameFilter, setNameFilter] = React.useState<string>('');
+
+  const filteredCommits = React.useMemo(
+    () =>
+      nameFilter ? commits.filter((commit) => commit.sha.indexOf(nameFilter) !== -1) : commits,
+    [nameFilter, commits],
+  );
+
+  const onClearFilters = () => setNameFilter('');
+  const onNameInput = (name: string) => setNameFilter(name);
+
+  const emptyMessage = (
+    <EmptyState variant={EmptyStateVariant.full}>
+      <EmptyStateIcon icon={SearchIcon} />
+      <Title headingLevel="h2" size="lg">
+        No results found
+      </Title>
+      <EmptyStateBody>
+        No results match the filter criteria. Remove filters or clear all filters to show results.
+      </EmptyStateBody>
+      <EmptyStateSecondaryActions>
+        <Button variant="link" onClick={onClearFilters} data-test-id="catalog-clear-filters">
+          Clear all filters
+        </Button>
+      </EmptyStateSecondaryActions>
+    </EmptyState>
+  );
+
+  return (
+    <PageSection padding={{ default: 'noPadding' }} variant={PageSectionVariants.light} isFilled>
+      <Title size="lg" headingLevel="h3" className="pf-c-title pf-u-mt-lg pf-u-mb-sm">
+        {recentOnly ? 'Recent commits' : 'Commits'}
+      </Title>
+      <Text className="pf-u-mb-lg">
+        Monitor your commits and their pipeline progression across all components
+        <Button variant="link" className="pf-u-pl-sm">
+          learn more
+        </Button>
+      </Text>
+      <Toolbar data-testid="component-list-toolbar" clearAllFilters={onClearFilters}>
+        <ToolbarContent>
+          <ToolbarGroup alignment={{ default: 'alignLeft' }}>
+            <ToolbarItem className="pf-u-ml-0">
+              <SearchInput
+                name="nameInput"
+                data-test="name-input-filter"
+                type="search"
+                aria-label="name filter"
+                placeholder="Filter by name..."
+                onChange={(name) => onNameInput(name)}
```

Comment:
After clearing the filter, the search term is not cleared. 
```suggestion
                onChange={(name) => onNameInput(name)}
                value={nameFilter}
```